### PR TITLE
refresh logic implemented

### PR DESCRIPTION
**Pull request by rotemkadoshvia**
eventually i'v decided to hold a member with the list Livedata and to observe it from both fragments

### CODE REVIEW

File: app/src/main/java/com/example/support/BindingAdapters.kt [hunk 1]
```diff
@@ -17,23 +17,20 @@
...
-import com.example.support.team.TeamMemberApiStatus
 import com.example.support.team.MemberViewAdapter
 
 
 @BindingAdapter("listData")
-fun <T>bindRecyclerView(recyclerView: RecyclerView, data: List<TeamMember>?) {
+fun bindRecyclerView(recyclerView: RecyclerView, data: List<TeamMember>?) {
     val adapter = recyclerView.adapter as MemberViewAdapter
-    adapter.submitList(data)
+    adapter.submitList(data as MutableList<TeamMember>?)
```
Comment (noarazvia) [inline app/src/main/java/com/example/support/BindingAdapters.kt]: `data.toMutableList()`

File: app/src/main/java/com/example/support/RefreshViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import androidx.annotation.DrawableRes
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.support.networking.MembersApi
+import com.example.support.networking.TeamMember
+import kotlinx.coroutines.launch
+
+data class TeamMemberApiStatus(var isVisible: Boolean, @DrawableRes var imageResource: Int)
```
Comment (noarazvia) [inline app/src/main/java/com/example/support/RefreshViewModel.kt:13]: can these be `val` ?

File: app/src/main/java/com/example/support/member/MemberFragment.kt [hunk 3]
```diff
@@ -2,53 +2,65 @@
...
+import com.example.support.RefreshViewModel
 import com.example.support.databinding.MemberFragmentBinding
 import com.example.support.networking.TeamMember
 import com.squareup.picasso.Picasso
-import kotlinx.android.synthetic.main.member_fragment.view.*
 
 class MemberFragment : Fragment() {
     private lateinit var binding: MemberFragmentBinding
+    private val refreshViewModel: RefreshViewModel by activityViewModels()
+    private lateinit var viewModel : MemberViewModel
```
Comment (noarazvia) [inline app/src/main/java/com/example/support/member/MemberFragment.kt]: lazy initialisation also

**Comment by noarazvia** [comment]
I started going through this, and saw a lot of code that was already merged. it looked strange, and now I see that the base branch is incorrect.
lets delete remote master and change the default branch to `master-new`

File: app/src/main/java/com/example/support/RefreshViewModel.kt (hunk 2 above)
Comment (rotemkadoshvia) [reply app/src/main/java/com/example/support/RefreshViewModel.kt:13]: yes, i will change

**Comment by RotemKadosh** [comment]
I see the mess i made, changed the default branch to be master_new

File: app/src/main/java/com/example/support/team/MemberViewAdapter.kt [hunk 4]
```diff
@@ -32,7 +32,10 @@ class MemberViewAdapter(private val onClickListener: OnClickListener) :
         }
 
         override fun areContentsTheSame(oldItem: TeamMember, newItem: TeamMember): Boolean {
-            return (oldItem.available == newItem.available && oldItem.imgSrcUrl == newItem.imgSrcUrl)
+            return (oldItem.available == newItem.available &&
```
Comment (noarazvia) [inline app/src/main/java/com/example/support/team/MemberViewAdapter.kt]: use equals

**Review by noarazvia — APPROVED** [review]

**Pull request merged**